src/components/TimePicker/timeSelect.js

```diff
@@ -342,7 +351,7 @@ export default class TimeSelect extends Component {
             ampm,
         });
 
-        if (currentHour && currentMinutes && ampm) {
+        if (currentHour && currentMinutes) {
```

Comment:
Here we need a check because in the case of 12 hours if the user doesn't select the am pm value then we should not send the onChange event.